### PR TITLE
data model: implement seriesStatement, partOf and issuance transformation

### PR DESCRIPTION
**Pull request by reropag**
Theses new fields are seriesStatement, partOf and issuance
transformation.

* Implements transformation for Marc21.
* Implements transformation for UNIMARC.
* Adapts ES mapping.
* Adapts JSON schemas.

Co-Authored-by: Gianni Pante <gianni.pante@rero.ch>

## Why are you opening this PR?

task: https://tree.taiga.io/project/rero21-reroils/task/1496?kanban-status=1224894
US: https://tree.taiga.io/project/rero21-reroils/us/1432?milestone=268117

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* https://github.com/rero/rero-ils/pull/1046
* https://github.com/rero/rero-ils-ui/pull/283

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json [hunk 1]
```diff
@@ -282,6 +358,113 @@
         }
       }
     },
+    "issuance": {
+      "title": "Mode of issuance",
+      "type": "object",
+      "required": [
+        "maintype",
```
Comment (AoNoOokami) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json:365]: mainType?

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json [hunk 2]
```diff
@@ -282,6 +358,113 @@
...
+        "subtype"
```
Comment (AoNoOokami) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json:366]: subType?

File: rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py [hunk 3]
```diff
@@ -885,15 +923,186 @@ def marc21_to_notes(self, key, value):
...
+        The form 'lastname, firstname' of the author form subfield a
+        is a appended to the subfield_t in the following form:
+        ' / firstname lastname'
+        """
+        items = get_field_items(value)
+        new_data = []
+        author = None
+        pending_g_values = []
+        pending_v_values = []
+        index = 0
```
Comment (rerowep) [inline rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py:1008]: Can not see where `index` is used in next code.

File: rero_ils/modules/documents/dojson/contrib/unimarctojson/model.py [hunk 4]
```diff
@@ -395,20 +456,46 @@ def unimarc_description(self, key, value):
...
-    if number:
-        series['number'] = ', '.join(utils.force_list(number))
-    return series
+    # normalize the value by building couple of $a, $v
+    # $v can be missing  in a couple
+    items = get_field_items(value)
+    new_data = []
+    fist_a_value = None
+    pending_v_values = []
+    index = 0
```
Comment (rerowep) [inline rero_ils/modules/documents/dojson/contrib/unimarctojson/model.py:471]: Can not see where `index` is used in next code.

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json [hunk 5]
```diff
@@ -863,41 +1046,92 @@
         }
       }
     },
-    "series": {
-      "title": "Series",
-      "description": "Series to which belongs the resource.",
+    "seriesStatement": {
+      "title": "Series statement",
+      "description": "Series and eventual subseries to which a resource belongs (MARC 490).",
```
Comment (rerowep) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json:1051]: Do we have to say it comes from MARC 490?

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json [hunk 6]
```diff
@@ -236,13 +237,100 @@
...
-    "is_part_of": {
-      "title": "Is part of",
-      "description": "Title of the host document.",
-      "type": "string",
-      "minLength": 1,
-      "form": {
-        "hide": true
+    "partOf": {
+      "title": "Part of (host document)",
+      "description": "Host document, for example a journal or a series (MARC 773, 800, 830).",
```
Comment (rerowep) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json:242]: Do we have to say it comes from MARC?

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json (hunk 1 above)
Comment (iGormilhit) [reply rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json:365]: @AoNoOokami @reropag https://github.com/rero/developer-resources/blob/master/guidelines/json-schemas.md#key-names

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json [hunk 7]
```diff
@@ -232,13 +233,88 @@
...
+        "properties": {
+          "document": {
+            "title": "Host document",
+            "type": "object",
+            "required": [
+              "$ref"
+            ],
+            "properties": {
+              "$ref": {
+                "title": "Document Link",
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json:261]: Change `Document Link` to `Document link`.

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json [hunk 8]
```diff
@@ -282,6 +358,113 @@
...
+        "maintype",
+        "subtype"
+      ],
+      "propertiesOrder": [
+        "maintype",
+        "subtype"
+      ],
+      "additionalProperties": false,
+      "properties": {
+        "maintype": {
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json:374]: Same as other comment of @AoNoOokami: `mainType` or `main_type`?
https://github.com/rero/developer-resources/blob/master/guidelines/json-schemas.md#key-names

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json [hunk 9]
```diff
@@ -282,6 +358,113 @@
...
+                "label": "multipart monograph",
+                "value": "rdami:1002"
+              },
+              {
+                "label": "serial",
+                "value": "rdami:1003"
+              },
+              {
+                "label": "integrating resource",
+                "value": "rdami:1004"
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json:400]:
```suggestion
                "label": "Single unit",
                "value": "rdami:1001"
              },
              {
                "label": "Multipart monograph",
                "value": "rdami:1002"
              },
              {
                "label": "Serial",
                "value": "rdami:1003"
              },
              {
                "label": "Integrating resource",
                "value": "rdami:1004"
```

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json [hunk 10]
```diff
@@ -282,6 +358,113 @@
...
+                "label": "set",
+                "value": "set"
+              },
+              {
+                "label": "updating website",
+                "value": "updatingWebsite"
+              },
+              {
+                "label": "updating loose leaf",
+                "value": "updatingLoose-leaf"
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json:461]: Same as precedent comment, a capitalized initial to the first word of labels.

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json (hunk 5 above)
Comment (iGormilhit) [reply rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json:1051]: @rerowep Yes, this description comes from the metadata group specifications. It's there to help librarian, at least the ones that can speak MARC fluently.

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json (hunk 6 above)
Comment (iGormilhit) [reply rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json:261]: @rerowep Yes, this description comes from the metadata group specifications. It's there to help librarian, at least the ones that can speak MARC fluently.

You'll have to read the specification files… :smile:

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json (hunk 5 above)
Comment (reropag) [reply rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json:1051]: Yes

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json (hunk 6 above)
Comment (reropag) [reply rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json:261]: Yes

**Pull request merged**